src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,532 @@
+  {
+    // Create rendering camera
+    this->dataPtr->boundingboxCamera =
+      this->Scene()->CreateBoundingBoxCamera(this->Name());
+
+    this->dataPtr->camera = this->Scene()->CreateCamera(
+      this->Name() + "_rgbCamera");
+  }
+
+  auto width = sdfCamera->ImageWidth();
+  auto height = sdfCamera->ImageHeight();
+
+  // Set Camera Properties
+  this->dataPtr->camera->SetImageFormat(rendering::PF_R8G8B8);
+  this->dataPtr->camera->SetImageWidth(width);
+  this->dataPtr->camera->SetImageHeight(height);
+  this->dataPtr->camera->SetVisibilityMask(sdfCamera->VisibilityMask());
+  this->dataPtr->camera->SetNearClipPlane(sdfCamera->NearClip());
+  this->dataPtr->camera->SetFarClipPlane(sdfCamera->FarClip());
+  this->dataPtr->camera->SetNearClipPlane(0.01);
+  this->dataPtr->camera->SetFarClipPlane(1000);
+  math::Angle angle = sdfCamera->HorizontalFov();
+  if (angle < 0.01 || angle > IGN_PI*2)
+  {
+    ignerr << "Invalid horizontal field of view [" << angle << "]\n";
+    return false;
+  }
+  double aspectRatio = static_cast<double>(width)/height;
+  this->dataPtr->camera->SetAspectRatio(aspectRatio);
+  this->dataPtr->camera->SetHFOV(angle);
+
+  this->dataPtr->boundingboxCamera->SetImageWidth(width);
+  this->dataPtr->boundingboxCamera->SetImageHeight(height);
+  this->dataPtr->boundingboxCamera->SetNearClipPlane(sdfCamera->NearClip());
+  this->dataPtr->boundingboxCamera->SetFarClipPlane(sdfCamera->FarClip());
+  this->dataPtr->boundingboxCamera->SetImageFormat(PixelFormat::PF_R8G8B8);
+  this->dataPtr->boundingboxCamera->SetAspectRatio(aspectRatio);
+  this->dataPtr->boundingboxCamera->SetHFOV(angle);
+  this->dataPtr->boundingboxCamera->SetVisibilityMask(
+    sdfCamera->VisibilityMask());
+  this->dataPtr->boundingboxCamera->SetBoundingBoxType(this->dataPtr->type);
+
+  // Add the camera to the scene
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->camera);
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->boundingboxCamera);
+
+  // Add the rendering sensor to handle its render, no need to add the
+  // rgb camera as we handle its render via Capture() function
+  this->AddSensor(this->dataPtr->boundingboxCamera);
+
+  // Connection to receive the BoundingBox buffer
+  this->dataPtr->newBoundingBoxConnection =
+    this->dataPtr->boundingboxCamera->ConnectNewBoundingBoxes(
+      std::bind(&BoundingBoxCameraSensor::OnNewBoundingBoxes, this,
+        std::placeholders::_1));
+
+  this->dataPtr->image = this->dataPtr->camera->CreateImage();
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+rendering::BoundingBoxCameraPtr BoundingBoxCameraSensor::BoundingBoxCamera()
+{
+  return this->dataPtr->boundingboxCamera;
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensor::OnNewBoundingBoxes(
+  const std::vector<rendering::BoundingBox> &boxes)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  this->dataPtr->boundingBoxes.clear();
+  for (auto box : boxes)
```

Comment:
```suggestion
  for (const auto &box : boxes)
```
nit: use a reference to avoid a copy. I also believe you can use a `const` reference since we aren't changing the bounding box here